### PR TITLE
[Bots][Reviewedandupdatedcontent][1427100]

### PR DESCRIPTION
**Pull request by v-rodriguezd**
Changes to URL and grammar and language changes. Awaiting inputs from @Trinetra-MSFT for a screenshot.

### CODE REVIEW

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 1]
```diff
@@ -2,97 +2,164 @@
...
 
-The request to get the token is a normal POST message request (using the existing message schema). It is included in the attachments of an OAuthCard. The schema for the OAuthCard class is defined in [Microsoft Bot Schema 4.0](/dotnet/api/microsoft.bot.schema.oauthcard?view=botbuilder-dotnet-stable&preserve-view=true) and it is very similar to a sign-in card. Teams will treat this request as a silent token acquisition if the `TokenExchangeResource` property is populated on the card. For the Teams channel, we honor only the `Id` property, which uniquely identifies a token request.
+#### Request a bot token
+
+The request to get the token is a normal POST message request (using the existing message schema). It is included in the attachments of an OAuthCard. The schema for the OAuthCard class is defined in [Microsoft Bot Schema 4.0](/dotnet/api/microsoft.bot.schema.oauthcard?view=botbuilder-dotnet-stable&preserve-view=true) and it is similar to a sign-in card. Teams treats this request as a silent token acquisition if the `TokenExchangeResource` property is populated on the card. For the Teams channel, we honor only the `Id` property, which uniquely identifies a token request.
```
Comment (KirtiPereira) [inline msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md:144]:
```Suggestion
The request to get the token is a normal POST message request (using the existing message schema). It is included in the attachments of an OAuthCard. The schema for the OAuthCard class is defined in [Microsoft Bot Schema 4.0](/dotnet/api/microsoft.bot.schema.oauthcard?view=botbuilder-dotnet-stable&preserve-view=true) and it is similar to a sign-in card. Teams treats this request as a silent token acquisition if the `TokenExchangeResource` property is populated on the card. For the Teams channel, only the `Id` property, which uniquely identifies a token request, is honored.
```

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 2]
```diff
@@ -2,97 +2,164 @@
...
 
 >[!NOTE]
-> The Bot Framework `OAuthPrompt` or the `MultiProviderAuthDialog` is supported for single sign-on (SSO) authentication.
+> The Microsoft Bot Framework `OAuthPrompt` or the `MultiProviderAuthDialog` is supported for SSO authentication.
 
-If this is the first time the user is using your application and the user consent is required, the user will be shown a dialog to continue with the consent experience similar to the one below. When the user selects **Continue**, two different things occur depending on whether the bot is defined or not and a sign-in button on the OAuthCard.
+If the user is using your application for the first time and user consent is required, the following dialog box is shown to continue with the consent experience:
```
Comment (KirtiPereira) [inline msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md:149]:
```Suggestion
If the user is using the application for the first time and user consent is required, the following dialog box appears to continue with the consent experience:
```

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 3]
```diff
@@ -2,97 +2,164 @@
...
 
 ![Consent dialog box](../../../assets/images/bots/bots-consent-dialogbox.png)
 
-If the bot defines a sign-in button, the sign-in flow for bots will be triggered similarly to the sign-in flow from a card button in a message stream. It is up to the developer to decide which permissions to ask for the user to consent. This approach is recommended if you need a token with permissions beyond `openId`, for example, if you want to exchange the token for graph resources.
+When the user selects **Continue**, two different things occur depending on if the bot is defined or not, and a sign-in button appears on the OAuthCard.
```
Comment (KirtiPereira) [inline msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md:153]: As discussed, please get clarity on the sentence.

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 4]
```diff
@@ -2,97 +2,164 @@
...
+
+If the bot defines a sign-in button, the sign in flow for bots is triggered similar to the sign in flow from a card button in a message stream. It is up to the developer to decide which permissions require user's consent. This approach is recommended if you need a token with permissions beyond `openId`. For example, if you want to exchange the token for graph resources.
```
Comment (KirtiPereira) [inline msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md:155]:
```Suggestion
If the bot defines a sign-in button, the sign in flow for bots is triggered similar to the sign in flow from a card button in a message stream. The developer must decide which permissions require user's consent. This approach is recommended if you require a token with permissions beyond `openId`. For example, if you want to exchange the token for graph resources.
```

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 5]
```diff
@@ -108,85 +175,62 @@ var attachment = new Attachment
...
 ```
 
-#### Receiving the token
+#### Receive the bot token
 
-The response with the token is sent through an invoke activity with the same schema as others invoke activities the bots receive today. The only difference is the invoke name, **sign-in/tokenExchange** and the **value** field which will contain the **Id** (a string) of the initial request to get the token and the **token** field (a string value including the token). Please note that you might receive multiple responses for a given request if the user has multiple active endpoints. It is up to you to deduplicate the responses with the token.
+The response with the token is sent through an invoke activity with the same schema as other invoke activities that the bots receive today. The only difference is the invoke name, **sign-in/tokenExchange** and the **value** field, which contains the **Id** (a string) of the initial request to get the token and the **token** field (a string value including the token).
+
+>[!NOTE]
+> You might receive multiple responses for a given request if the user has multiple active endpoints. It is up to you to deduplicate the responses with the token.
```
Comment (KirtiPereira) [inline msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md:186]:
```Suggestion
> You might receive multiple responses for a given request if the user has multiple active endpoints. You must deduplicate the responses with the token.
```

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 6]
```diff
@@ -108,85 +175,62 @@ var attachment = new Attachment
...
-    >[!NOTE]
-    > **Implicit grant** may be required in the AAD application.
-
-    >* For the Token Exchange URL, use the scope value defined in the previous step of your AAD application. The presence of the Token Exchange URL is indicating to the SDK that this AAD application is configured for SSO.
-    >* Specify "common" as the **Tenant ID**.
-    >* Add all the scopes configured when specifying permissions to downstream APIs for your AAD application. With the client id and client secret provided, the token store will exchange the token for a graph token with defined permissions for you.
-    >* Select **Save**.
-
-    ![VuSSOBotConnection setting view](../../../assets/images/bots/bots-vuSSOBotConnection-settings.png)
+The `turnContext.activity.value` is of type [TokenExchangeInvokeRequest](/dotnet/api/microsoft.bot.schema.tokenexchangeinvokerequest?view=botbuilder-dotnet-stable&preserve-view=true) and contains the token that can be further used by your bot. You need to store the tokens for performance reasons and refresh them.
```
Comment (KirtiPereira) [inline msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md:213]:
```Suggestion
The `turnContext.activity.value` is of type [TokenExchangeInvokeRequest](/dotnet/api/microsoft.bot.schema.tokenexchangeinvokerequest?view=botbuilder-dotnet-stable&preserve-view=true) and contains the token that can be further used by your bot. You must store the tokens for performance reasons and refresh them.
```

**Review by KirtiPereira — CHANGES_REQUESTED** [review]
@v-rodriguezd - please verify and commit the suggestions.

**Comment by v-rodriguezd** [comment]
> @v-rodriguezd - please verify and commit the suggestions.

Screenshot for Request a bot token section currently not available and will be updated later.

File: msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md [hunk 7]
```diff
@@ -2,97 +2,164 @@
...
+2. When the current user uses your bot application for the first time, there is a request prompt to provide consent (if consent is required) or to handle step-up authentication (such as two-factor authentication).
 
-3. Microsoft Teams requests the bot application token from the Azure AD endpoint for the current user.
+3. Teams requests the bot application token from the AAD endpoint for the current user.
 
-4. Azure AD sends the bot application token to the Teams application.
+4. AAD sends the bot application token to the Teams application.
 
-5. Microsoft Teams sends the token to the bot as part of the value object returned by the invoke activity with the name sign-in/tokenExchange.
+5. Teams sends the token to the bot as part of the value object returned by the invoke activity with the name sign-in/tokenExchange.
```
[the remark this answers is not in the document]
Comment (v-rodriguezd) [reply msteams-platform/bots/how-to/authentication/auth-aad-sso-bots.md]: sign-in/tokenExchange is a name of function so leaving as is and making it bold as it is a function name

**Review by Trinetra-MSFT — APPROVED** [review]
Looks Good

**Review by KirtiPereira — APPROVED** [review]
Approved for merging

**Review by surbhigupta12 — APPROVED** [review]
Approving the file.

**Pull request merged**